packages/graph/hash_graph/lib/graph/src/store/postgres/query/expression/join.rs

```diff
@@ -0,0 +1,117 @@
+use std::fmt;
+
+use crate::store::postgres::query::{Condition, Expression, Table, Transpile};
+
+#[derive(Debug, PartialEq, Eq, Hash)]
+pub struct JoinExpression<'q> {
+    pub table: Table,
+    pub on: Condition<'q>,
+}
+
+impl<'q> JoinExpression<'q> {
+    #[must_use]
+    pub const fn from_tables(table: Table, on: Table) -> Self {
+        Self {
+            table,
+            on: Condition::Equal(
+                Some(Expression::Column(table.source_join_column())),
+                Some(Expression::Column(on.target_join_column())),
+            ),
+        }
+    }
+}
+
+impl Transpile for JoinExpression<'_> {
+    fn transpile(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        fmt.write_str("JOIN ")?;
+        if self.table.alias.is_some() {
+            let unaliased_table = Table {
+                name: self.table.name,
+                alias: None,
```

Comment:
Are we doing this to still make use of transpile to generate the name, regardless of the missing alias, through `transpile`?

Reply:
This ensures, that the join is either `JOIN table ON ...` or `JOIN table AS table_x_y ON`. So, if `table` has an alias, the plain name + ` AS ` will be printed as well.